src/v2i-hub/ImmediateForwardPlugin/src/ImmediateForwardPlugin.cpp

```diff
@@ -271,122 +137,107 @@ namespace ImmediateForward
+							c = ::tolower(c);
+						});
+						/* convert to hex array */
 
-					string cmd1="curl -X POST "+url+" -H \'Content-Type: application/json\' -d "+req; 
-					const char *cmd=cmd1.c_str();  
-					char buffer[2048];
-					std::string result="";
-					FILE* pipe= popen(cmd,"r"); 
+						string msgString=msg->payload->valuestring;
+						string base64str=""; 
 
-					if (pipe == NULL ) 
-						throw std::runtime_error("popen() failed!");
-					try{
-						while (fgets(buffer, sizeof(buffer),pipe) != NULL)
-						{
-							result+=buffer; 
-						}
-					} catch (std::exception const & ex) {
-						
-						pclose(pipe);
-						SetStatus<uint>(Key_SkippedSignError, ++_skippedSignErrorResponse);
-						PLOG(logERROR) << "Error parsing Messages: " << ex.what();
-						return;
-					}
-					PLOG(logDEBUG1) << "SCMS Contain response = " << result << std::endl;
-					cJSON *root   = cJSON_Parse(result.c_str());
-					// Check if status is 200 (successful)
-					cJSON *status = cJSON_GetObjectItem(root, "code");
-					if ( status ) {
-						// IF status code exists this means the SCMS container returned an error response on attempting to sign
-						// Set status will increment the count of message skipped due to signature error responses by one each
-						// time this occurs. This count will be visible under the "State" tab of this plugin.
-						cJSON *message = cJSON_GetObjectItem(root, "message");
-						SetStatus<uint>(Key_SkippedSignError, ++_skippedSignErrorResponse);
-						PLOG(logERROR) << "Error response from SCMS container HTTP code " << status->valueint << "!\n" << message->valuestring << std::endl;
-						return;
-					}
-					cJSON *sd = cJSON_GetObjectItem(root, "signedMessage");
-					string signedMsg = sd->valuestring;
-					base642hex(signedMsg,payloadbyte); // this allows sending hex of the signed message rather than base64
+						hex2base64(msgString,base64str);  
 
-				}
-				else 
-				{
-					payloadbyte=msg->payload->valuestring; 
-				}
-				// @SONAR_START@
+						std::string req = "\'{\"type\":\""+mType+"\",\"message\":\""+base64str+"\"}\'";
 
-				os << "Version=0.7" << "\n";
-				os << "Type=" << _messageConfigMap[configIndex].SendType << "\n" << "PSID=" << _messageConfigMap[configIndex].Psid << "\n";
-				if (_messageConfigMap[configIndex].Channel.empty())
-					os << "Priority=7" << "\n" << "TxMode=" << txMode << "\n" << "TxChannel=" << msg->dsrcMetadata->channel << "\n";
-				else
-					os << "Priority=7" << "\n" << "TxMode=" << txMode << "\n" << "TxChannel=" << _messageConfigMap[configIndex].Channel << "\n";
-				os << "TxInterval=0" << "\n" << "DeliveryStart=\n" << "DeliveryStop=\n";
-				os << "Signature=" << (signState == 1 ? "True" : "False") << "\n" << "Encryption=False\n";
-				os << "Payload=" << payloadbyte << "\n";
 
-				string message = os.str();
 
+						string cmd1="curl -X POST " + imfConfig.hsmUrl.value() + "sign" + " -H \'Content-Type: application/json\' -d "+req; 
```

Comment:
Is this hsmUrl constantly changing, why cannot we use constant value for the hsmUrl?